### DISCUSSION TITLE
Как правильно настроить на OpenWrt роутере?

### DISCUSSION THREAD

**Discussion by 0x131313** [hufrea/byedpi#206] [Q&A] [answered]
Помогите пожалуйста разобраться, как будто, что-то делаю не так.

**Вводные:**
1) на win10 работает **_goodbyedpi_** (там свой конфиг - просто для инфы)
2) на android работает **_ByeDPIAndroid_** ( `-n "www.google.com" -q 1+s -O 1 -s 25+s -t 5` )
3) на win10 работает - **_byedpi.bat_** ( `ciadpi.exe --ip 127.0.0.1 --port 18081 --split 2 --disorder 7`) + если прописать прокси в винде `socks=127.0.0.1 | port 18081`
4) на роутере с openwrt стоит сборка https://github.com/spvkgn/ByeDPI-OpenWrt/releases/tag/v0.14.1-23.05 - и на нем прописываю тот же конфиг от винды без IP (`--port 18081 --split 2 --disorder 7`) и НЕ работает.

**Вопросы:**
1) что делаю не так в п.4? (почему на андроиде и батник на винде - работает, а на роутере - нет)
2) может прокси надо прописывать где-то в роутере?

Спасибо!

**Comment by Miraihi** [comment]
Я никогда не занимался настройкой на OpenWRT, но гарантирую что к byedpi нужен прокси-клиент, так как сам по себе byedpi только создаёт прокси-сервер, но ничего к нему не подключает (Разве что на Android он сразу ведёт трафик через собственное VPN-решение, но это исключение).
Можете попробовать воспользоваться [Clash](https://github.com/chandelures/luci-app-simple-clash), хотя возможно есть более легковесные варианты.
Так как я сам планирую переходить на OpenWRT, меня эта тема также интересует.

**Reply by 0x131313** [reply]
спасибо за ответ!
вот хотелось бы узнать как и что поставить (легковесное, т.к. мало памяти на роутере) - для того чтобы закончить настройку.

**Reply by Miraihi** [reply]
Расширьте память с помощью exroot и обычной флешки, и о свободном месте можно будет не беспокоиться. Сам это делал раньше.
https://openwrt.org/docs/guide-user/additional-software/extroot_configuration

**Reply by 0x131313** [reply]
У роутера нет usb

**Reply by 0x131313** [reply]
Есть очень легковесный microsocks
Но пока не нашел какой либо инструкции как связать его и byedpi сервер, что запускается (как обсуждали выше)

**Reply by Miraihi** [reply]
[Tinyproxy](https://github.com/tinyproxy/tinyproxy) может подойти. Теоретически режим Transparent proxy должен быть как раз тем, что нужно. Я этим не занимался, опять-таки.
Ну, и ещё имейте в виду, что к byedpi можно подключаться за пределами роутера, просто введя в прокси-клиент на устройствах подключённых пользователей `IP роутера:порт byedpi`. Например вот как это выглядит у меня в windows 11.
![ApplicationFrameHost_wNTxZ3svoC](https://github.com/user-attachments/assets/227c7ded-1f89-4c4a-9307-d4f71c3f251c)

**Reply by 0x131313** [reply]
спасибо за ответы!
1) да, tinyproxy рассматривал - но я так понял он не про socks - или я ошибаюсь?

2) про вариант подключения к прокси с винды - да, у меня была аналогичная мысль (т.к. виндовый byedpi так работает) и пробовал этот вариант и поробовал ещё раз. Оно вроде работает (интернет работает) - потому как если указать неправильный порт - доступа в инет с винды нет, а если правильный - то есть. Но ютуб не открывается - конфиг аналогичный виндовому (рабочему). Как только запускаем локально byedpi - и прописываем так же прокси в настройках - ютуб заводится.

Сам конфиг на роутере такой:
`/usr/bin/ciadpi --port 18081 --split 2 --disorder 7`

На винде, он же + `--ip 127.0.0.1` (а можно и без порта и ip)

И вот тут пока мне непонятно в чем именно проблема.
___
пробовал указать `--transparent` - прозрачный прокси, при котором socks не работает и если его пытаться указать в прокси, то инета нет (что логично). При этом режиме, насколько я его понимаю, весь трафик идущий через роутер должен идти через byedpi - но инет работает так же - обычные сайты работают, а ютуб - нет.
___
пробовал с телефона через nekob - создать в ручную профиль сокса5 и там по ипу роутера и порту - проверка этого подключения - не проходит.

**Reply by 0x131313** [reply]
Через винду и БЕЗ указания прокси в настройках винды:
```
curl -I -x socks5://192.168.1.44:18081 -k https://rutracker.org/myip
curl: (97) Could not resolve host: rutracker.org

curl -I -x socks5://192.168.1.44:18081 -k https://ya.ru
HTTP/1.1 302 Moved temporarily
...
```

Через винду и С указанием прокси в настройках винды:
```
curl -I -x socks5://192.168.1.44:18081 -k https://rutracker.org/myip
curl: (35) Recv failure: Connection was reset

curl -I -x socks5://192.168.1.44:18081 -k https://ya.ru
HTTP/1.1 302 Moved temporarily
...
```

**Reply by Miraihi** [reply]
> спасибо за ответы!
> 
> 1. да, tinyproxy рассматривал - но я так понял он не про socks - или я ошибаюсь?
> 2. про вариант подключения к прокси с винды - да, у меня была аналогичная мысль (т.к. виндовый byedpi так работает) и пробовал этот вариант и поробовал ещё раз. Оно вроде работает (интернет работает) - потому как если указать неправильный порт - доступа в инет с винды нет, а если правильный - то есть. Но ютуб не открывается - конфиг аналогичный виндовому (рабочему). Как только запускаем локально byedpi - и прописываем так же прокси в настройках - ютуб заводится.
> 
> Сам конфиг на роутере такой: `/usr/bin/ciadpi --port 18081 --split 2 --disorder 7`
> 
> На винде, он же + `--ip 127.0.0.1` (а можно и без порта и ip)
> 
> И вот тут пока мне непонятно в чем именно проблема.
> 
> пробовал указать `--transparent` - прозрачный прокси, при котором socks не работает и если его пытаться указать в прокси, то инета нет (что логично). При этом режиме, насколько я его понимаю, весь трафик идущий через роутер должен идти через byedpi - но инет работает так же - обычные сайты работают, а ютуб - нет.
> 
> пробовал с телефона через nekob - создать в ручную профиль сокса5 и там по ипу роутера и порту - проверка этого подключения - не проходит.

1. Да, вы правы, поддержки Socks нет. Придётся искать дальше.
2. Могу предложить поиграть со [Clash for Windows](https://github.com/lantongxue/clash_for_windows_pkg) - это отдельный системный прокси-клиент. Есть довольно удобный графический интерфейс, может получится понять в чём проблема.

Наверное мои знания на этом исчерпаны, и пока я сам не попробую это настроить больше ничего полезного не скажу.
Разве что упомяну, что ваша стратегия для youtube рабочая, но не оптимальная (Страдает от долгих подзагрузок перед видео и не лучшей производительности). У меня сейчас практически идеально работает `ciadpi --hosts blacklist.txt --proto=https,tlc --fake -2 --ttl 3 --tls-sni "www.google.com" --tlsrec 27+s`. --ttl только надо подбирать индивидуально, минимально рабочее значение.
Разумеется, со временем РКН даже такую стратегию могут победить, так что надо держать руку на пульсе.

**Reply by 0x131313** [reply]
Спасибо вам за помощь, мне помогли вот тут (https://github.com/hufrea/byedpi/discussions/195#discussioncomment-10987961)

Возможно вам будет полезно.

Про стратегию - спасибо, тоже попробую и да, держим на пульсе ;)

**Comment by vehametra** [comment]
У меня ciadpi  не заработал в виде службы.
В итоге  так запускается
```
CMD="ciadpi --debug 1 --split 1 --disoob 3 --disorder 7 --fake -2 --tls-sni www.google.com --ip-opt --tlsrec=6+s --tlsrec 27+s --tlsrec 3+s --mod-http d  -F -H /root/hosts.txt"
ssh $WRT "tmux new-session -d -s \"000\" ${CMD} "

```

**Reply by 0x131313** [reply]
Если правильно вас понял, вы описываете запуск на винде, верно? 
_(если да, то с виндой нет проблем - я описал как там работает)_

Моя текущая проблема-недонастройка-вопрос - про роутер, где установлен byedpi, но не настроен прокси. Пока не нашел гайда, как это сделать с минимальными затратами по занимаемой памяти

**Reply by vehametra** [reply]
![Снимок экрана_2024-10-18_12-08-40](https://github.com/user-attachments/assets/a0180c12-13f5-4f62-82a3-309c11e6aacb)
вот эта версия. На опенврт. 
Запускаю скриптом с другой машины, из скрипта сервиса openwrt не заработало TFO и не вполне удобно опции подбирать, кроме это спамит активно в syslog.

**Comment by iceoeu** [comment]
Я установил zapret на openwrt. Отлично работает с конфигурацией:

NFQWS_OPT_DESYNC="--dpi-desync=fake,split --dpi-desync-fooling=md5sig,badseq --wssize 1:6 --dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin"

https://github.com/bol-van/zapret/blob/master/docs/readme.txt

**Reply by 0x131313** [reply]
Запрет - ок, но не влезет на мою модель.

**Reply by vehametra** [reply]
> Запрет - ок, но не влезет на мою модель.

Есть USB на роутере? 
ciadpi кстати должна влезать даже на mr3020.
Ну и про tmpfs  забывать не стоит.

**Reply by iceoeu** [reply]
Есть ещё такой вариант 
https://github.com/Waujito/youtubeUnblock?tab=readme-ov-file#youtubeunblock

**Comment by 0x131313** [comment, marked as answer]
Правильный ответ (thx @maximilionus):

> byedpi на роутере поднимает socks5 прокси-сервер, к которому надо либо подключаться через софт у клиентов (nekobox на PC и Android, Potatso на iOS), либо туннелировать весь входящий трафик внутри самого роутера.
